projects/angularplatform-server-ngcc/custom-test.js

```diff
@@ -0,0 +1,45 @@
+const {exec, execSync} = require('child_process');
+const {readFileSync} = require('fs');
+const {basename} = require('path');
+
+
+const projectName = basename(__dirname);
+const generatedIndexPath = `dist/${projectName}/browser/index.html`;
+const expectedIndexContent = `Welcome to ${projectName}!`;
+
+process.chdir(`${__dirname}/../..`);
+
+// Build the app and prerender `index.html` with SSR.
+// (This causes `@angular/platform-server` to be processed by ngcc.)
+run(`yarn ${projectName}:prerender`);
+
+// Verify that the generated `index.html` is pre-populated with the correct content.
+// (This ensures that prerendering works correctly.)
+const generatedIndexContent = readFileSync(generatedIndexPath, 'utf8');
+if (generatedIndexContent.includes(expectedIndexContent)) {
+  console.log('\'index.html\' prerendered correctly.');
+} else {
+  throw new Error(`Expected '${generatedIndexPath}' to contain '${expectedIndexContent}'.`);
+}
+
+// Run the e2e tests using the SSR server.
+// (This ensures that SSR works correctly.)
```

Comment:
Note: in here you are not testing SSR'd view but rather the client view. This tests though verifies that the server did respond successfully.

Reply:
Yeah, that it what I meant. I will re-word it to make it more clear.